packages/hop-node/src/theGraph/getTransferIds.ts

```diff
@@ -46,34 +46,26 @@ export default async function getTransferIds (
     token,
     startDate: 0,
     endDate: MaxInt32,
-    orderDirection: 'desc',
-    skip
+    lastId
   }
   if (filters.startDate) {
     variables.startDate = DateTime.fromISO(filters.startDate).toSeconds() >>> 0
   }
   if (filters.endDate) {
     variables.endDate = DateTime.fromISO(filters.endDate).toSeconds() >>> 0
   }
-  if (typeof filters.orderDesc === 'boolean') {
-    variables.orderDirection = filters.orderDesc ? 'desc' : 'asc'
-  }
   const jsonRes = await makeRequest(chain, query, variables)
   let transfers = jsonRes.transferSents.map((x: any) => normalizeEntity(x))
 
-  if (transfers.length === 1000) {
-    try {
-      transfers = transfers.concat(await getTransferIds(
-        chain,
-        token,
-        filters,
-        skip + 1000
-      ))
-    } catch (err) {
-      if (!err.message.includes('The `skip` argument must be between')) {
-        throw err
-      }
-    }
+  lastId = transfers[transfers.length - 1].id
```

Comment:
is it possible that `transfers.length` here can be `0` so `.id` would throw an error trying to read from `undefined`?

Reply:
Yes, great call. Fixed with 26ffbf5f93e5a0d1cd10ed6b2e87d8b1919e4177